ui/packages/code/interactive/Download.svelte

```diff
@@ -0,0 +1,91 @@
+<script lang="ts">
+	import { onDestroy } from "svelte";
+	import { fade } from "svelte/transition";
+	import { Download, Check } from "@gradio/icons";
+
+	export let value: string;
+	export let language: string;
+
+	$: ext = get_ext_for_type(language);
+
+	function get_ext_for_type(type: string) {
+		console;
+		const exts: Record<string, string> = {
+			py: "py",
+			python: "py",
+			md: "md",
+			markdown: "md",
+			json: "json",
+			html: "html",
+			css: "css",
+			js: "js",
+			javascript: "js",
+			ts: "ts",
+			typescript: "ts",
+			yaml: "yaml",
+			yml: "yml",
+			dockerfile: "dockerfile",
+			sh: "sh",
+			shell: "sh",
+			r: "r"
+		};
+
+		return exts[type] || "txt";
+	}
+
+	let copied = false;
+	let timer: NodeJS.Timeout;
+
+	function copy_feedback() {
+		copied = true;
+		if (timer) clearTimeout(timer);
+		timer = setTimeout(() => {
+			copied = false;
+		}, 2000);
+	}
+
+	$: download_value = URL.createObjectURL(new Blob([value]));
+
+	onDestroy(() => {
+		if (timer) clearTimeout(timer);
+	});
+</script>
+
+<a
+	download="file.{ext}"
+	href={download_value}
+	class:copied
+	on:click={copy_feedback}
+>
+	<Download />
+	{#if copied}
+		<span class="check" transition:fade><Check /></span>
+	{/if}
+</a>
+
+<style>
+	a {
+		position: relative;
+		cursor: pointer;
+		padding: 5px;
+
+		width: 22px;
+		height: 22px;
+	}
+
+	.copied {
+		color: var(--color-green-500);
```

Comment:
same comment as above

Reply:
swapped to neutral 'background' colour. orange was too 'warningy'.